var/spack/repos/builtin/packages/caliper/package.py

```diff
@@ -126,6 +126,11 @@ def cmake_args(self):
             args.append('-DMPI_CXX_COMPILER=%s' % spec['mpi'].mpicxx)
 
         if '+cuda' in spec:
+            # this avoids the CMake install step erasing the RPATH entry
+            # pointing to /path/to/cuda/extras/CUPTI/lib64, which is correctly
+            # added by Caliper's CMake but is not known to Spack. See upstream
+            # Spack issue: https://github.com/spack/spack/issues/24724
+            args.append('-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=On')
```

Comment:
don't have complete memory but in coreneuron we did something related: https://github.com/BlueBrain/CoreNeuron/blob/4c6e6aa96c4a981cd9c7eb185ce58a2ab29f4dec/CMake/config/SetRpath.cmake#L25